doc/design.md

```diff
@@ -0,0 +1,62 @@
+# Introduction
+
+## Acknowledgments
+
+This document's section structure is derived from the "[arc42](https://arc42.org/)" architectural template by Dr. Gernot Starke, Dr. Peter Hruschka.
+
+## Terms and Abbreviations
+
+<dl>
+    <dt>TDDB</dt><dd>Test Database Builder</dd>
+    <dt>Top-level database object</dt><dd>Database object that has no parent scope in the database.</dd>
+</dl>
+
+# Constraints
+
+This section introduces technical system constraints.
+
+# Solution Strategy
+
+TODO: add
```

Comment:
Do you want to merge the branch with TODO?

Reply:
https://github.com/exasol/test-db-builder-java/issues/10